.github/workflows/build.yml

```
@@ -133,7 +133,7 @@ jobs:
           tree -nh ./bin
       -
         name: Upload artifacts
-        uses: actions/upload-artifact@v4.3.0
+        uses: actions/upload-artifact@v4.3.4
```

we should use major only for `upload-artifact` and `download-artifact` in all workflows:

```suggestion
        uses: actions/upload-artifact@v4
```

see https://github.com/actions/download-artifact/issues/328#issuecomment-2209800401

---

I still think we should only use major as their API is often not backward compat